packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -776,6 +780,31 @@ export class Bucket extends BucketBase {
     return this.onEvent(EventType.ObjectRemoved, dest, ...filters);
   }
 
+  private validateBucketName(bucketName: string) {
```

Comment:
This is great, but I would appreciate if we could put the offending bucket name in the error messages, too... Like end with `(provided: "${bucketName}")`.

If you can make this fancier even and somehow highlight the offending part, this would be absolutely awesome... But may be challenging to make it "look right", so feel free not to.

Being said, it could be interesting to know the position of the offending element at least, due to the presence of unicode homoglyphs, for example:

OK | Not OK
---|---
`-` | `‐`, `－`
`a`| `ɑ`, `а`, `ａ`
`.` | `․`, `．`

Reply:
Thanks @RomainMuller 

Bucket name is in the error message. See line 804.

I'll include the offending character position in the error message, best effort.

Reply:
Added offending character offset in the error message. 

For each class of error, I only report the first offence that occur. 

The code has gotten slightly more complex with this change, although I've tried to keep it as readable as I could.